pkg/api/pod/util.go

```diff
@@ -55,6 +55,30 @@ func VisitContainers(podSpec *api.PodSpec, visitor ContainerVisitor) bool {
 	return true
 }
 
+// VisitContainersUnconditional is like VisitContainers, but it ignores feature gates for
+// new container types. This is necessary for things like detecting or dropping alpha
+// fields on an update when a feature-gated container has been added when the feature gate
+// was enabled and preserved even though the feature gate is now disabled. In this case
+// we still want to detect alpha fields within the now-feature-gated container.
+func VisitContainersUnconditional(podSpec *api.PodSpec, visitor ContainerVisitor) bool {
```

Comment:
Could you share more about why `VisitContainersUnconditional` is a new function instead of an option on `VisitContainers`?

Reply:
Sure, there are a few reasons:

1. @tedyu is building consensus in #82192 about how `VisitContainers` should change, and I think this should be a requirement in this discussion, but I don't want to block a bugfix on it.
2. I didn't want to tie `VisitContainers`'s signature to anything time-limited like specific feature gates.
3. Bug #82158 only affects a subset of uses of `VisitContainers`, so I wanted to avoid changing its signature prematurely.

I agree that this functionality should eventually make its way to `VisitContainers`, though.